### PR TITLE
Explictly set exported scalars in JSON API

### LINKED ISSUE #6805: Explicitly set exported fields in QuestionJsonPresenter
Some question types use `ContextualizedScalarsJsonPresenter` instead of a question-specific presenter. This exports all scalars in the question as strings, so if a new scalar is added to a question it will automatically be exported as a string.

Instead, we should explicitly set which fields we want exported and what types they should be. We can accomplish this by replacing the generic `ContextualizedScalarsJsonPresenter` with specific JsonPresenters for each question type.

### PR DESCRIPTION
**Pull request by avaleske**
### Description

Explicitly set exported scalars in the JSON API, instead of automatically exporting all scalars on a question.

- This prevents accidentally exposing a scalar in an internal format or a format that's not ideal for export, going forward.
- Scalars that were already exported in an in-ideal format, such as `serviceArea` are left unchanged and still exported as strings.
- Also alphabetizes the `case` statement in `QuestionJsonSampler`

## Release notes

n/a

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [n/a] Created unit and/or browser tests which fail without the change (if possible)
- [n/a] Performed manual testing (Chrome and Firefox if it includes front-end changes)
- [n/a] Extended the README / documentation, if necessary

### Issue(s) this completes

Fixes #6805

### CODE REVIEW

**Comment by avaleske** [comment]
* **#8037** <a href="https://app.graphite.dev/github/pr/civiform/civiform/8037?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @avaleske and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: server/app/services/export/QuestionJsonSampler.java [hunk 1]
```diff
@@ -184,7 +181,7 @@ public void addSampleData(ApplicantData applicantData, ApplicantQuestion applica
           /* latitude= */ 44.0462,
           /* longitude= */ -123.0236,
           /* wellKnownId= */ 4326L,
-          /* serviceArea= */ "springfield_county_InArea_1709069741,portland_NotInArea_1709069741");
+          /* serviceArea= */ "springfieldCounty_InArea_1709069741,portland_NotInArea_1709069741");
```
Comment (avaleske) [inline server/app/services/export/QuestionJsonSampler.java:184]: The underscore was causing issues deserializing, and now that we use the correct accessor methods we get back an actual list of `ServiceAreaInclusion`s we have to re-serialize for the API.

Comment (gwendolyngoetz) [reply server/app/services/export/QuestionJsonSampler.java:184]: Weird

**Review by gwendolyngoetz — APPROVED** [review]

File: server/app/services/export/QuestionJsonSampler.java (hunk 1 above)
Comment (avaleske) [reply server/app/services/export/QuestionJsonSampler.java:184]: Before the tests were just inserting a fake string directly into the `ApplicantData` and we were pulling it out as a string, so we were never having to serialize or deserialize anything. Now we pull it out as the correct type so we have to be able to deserialize it.

**Comment by avaleske** [comment]
### Merge activity

* **Jul 12, 6:28 PM EDT**: @avaleske merged this pull request with [Graphite](https://app.graphite.dev/github/pr/civiform/civiform/8037).

**Pull request merged**